File: src/main/java/shane/leetcode/problems/easy/Q237DeleteNodeInALinkedList.java
package shane.leetcode.problems.easy;

import com.tistory.shanepark.leetcode.ListNode;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class Q237DeleteNodeInALinkedList {

    @Test
    public void test() {
        ListNode node = ListNode.of(4, 5, 1, 9);
        deleteNode(node.next);
        assertThat(node).isEqualTo(ListNode.of(4, 1, 9));
    }

    public void deleteNode(ListNode node) {
        node.val = node.next.val;
        node.next = node.next.next;
    }
}

File: src/main/kotlin/shane/leetkode/problems/easy/Q237DeleteNodeInALinkedList.kt
package shane.leetkode.problems.easy

import com.tistory.shanepark.leetcode.ListNode
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class Q237DeleteNodeInALinkedList {

    @Test
    fun test() {
        val node = ListNode.of(4, 5, 1, 9)
        deleteNode(node.next)
        Assertions.assertThat(node).isEqualTo(ListNode.of(4, 1, 9))

    }

    fun deleteNode(node: ListNode) {
        node.`val` = node.next.`val`
        node.next = node.next.next
    }
}
